node/src/components/consensus/highway_core/state/panorama.rs

```diff
@@ -187,4 +187,29 @@ impl<C: Context> Panorama<C> {
         }
         Ok(())
     }
+
+    // Returns a bit mask that sets bit to 1 for every fault validator at equivocator's
+    // `ValidatorIndex`.
+    pub(crate) fn faults_bit_mask(&self) -> usize {
+        let mut bit_mask = 0;
+        for (idx, _) in self.enumerate().filter(|&(_, obs)| obs.is_faulty()) {
+            bit_mask |= 1 << idx.0;
+        }
+        bit_mask
+    }
+
+    /// Returns votes' hashes from this panorama that cite naively equivocator's vote.
+    pub(crate) fn need_endorsements(&self, state: &State<C>) -> Vec<&C::Hash> {
+        let local_faults = state.panorama.faults_bit_mask();
```

Comment:
```suggestion
        let local_faults = self.faults_bit_mask();
```

Reply:
I think this is correct, we want to check which of the cited votes are sent by equivocators (as seen by our local panorama).

Reply:
But our local state must not influence whether a particular vote is valid or not. And this is used in the validation code, too, isn't it?